crates/goose-cli/src/session/mod.rs

```diff
@@ -197,12 +197,20 @@ impl Session {
     /// Process a single message and get the response
     async fn process_message(&mut self, message: String) -> Result<()> {
         self.messages.push(Message::user().with_text(&message));
+        let is_first_message = self.messages.len() == 1;
 
         // Get the provider from the agent for description generation
         let provider = self.agent.provider().await;
 
         // Persist messages with provider for automatic description generation
+        // The first message creates the session file if it doesn't exist
+        // So we check that here and update the working directory after the first message
         session::persist_messages(&self.session_file, &self.messages, Some(provider)).await?;
+        if is_first_message {
+            let mut metadata = session::read_metadata(&self.session_file)?;
+            metadata.working_dir = std::env::current_dir().unwrap();
+            session::update_metadata(&self.session_file, &metadata).await?;
```

Comment:
this doesn't get set to my current working directory when just using `goose session` I just a get a default of `$HOME`

looking through it looks like `process_message` is called only if there's some initial message in an interactive session?

Reply:
only way i could get my `cwd` set was to do `./target/debug/goose run -s -t "hi"`, regular `goose s` didn't work for me